src/sqlfluff/core/templaters/slicers/tracer.py

```diff
@@ -165,6 +170,29 @@ def record_trace(self, target_slice_length, slice_idx=None, slice_type=None):
         if slice_type in ("literal", "templated"):
             self.source_idx += target_slice_length
 
+
+class JinjaAnalyzer:
+    """Analyzes a Jinja template to prepare for tracing."""
+
+    re_open_tag = regex.compile(r"^\s*({[{%])[\+\-]?\s*")
+    re_close_tag = regex.compile(r"\s*[\+\-]?([}%]})\s*$")
```

Comment:
Why are these not full Regexes - ie starting with ^ and ending with $?

Also wondering if gathering the whitespace before a tag has influence on indents (I note later on that search is used vs the lexed array) perhaps some of these matches are forced by what the lexer returns?

Reply:
The templated block is a whole string and not the individual components.

So `{% if ... %}` is all one templated string.

So we want `^{%` for opening tag (simplified), and `%}$` for closing tag (simplified) of that string, as @barrywhart has.

Reply:
In the past, we received one token at a time from Jinja, then combined them into a single string (`str_buff`) once they became a full block, e.g. `{% if a == 3 %}`. Now we still build `str_buff`, but we also keep the parts separate in an array `str_parts`. The regexes remain the same for legacy reasons. I _think_ it would work either way.

Mostly we want what the lexer returns, as the purpose of this class is to deduce exactly what Jinja is going to return for the "real" template, including tricky whitespace handling. That's been working pretty well for a while. If you look at `handle_left_whitespace_stripping()` and the code section (not function beginning with `# Right whitespace was stripped.`, you'll see how some of the trickier bits work.

There is _one_ case where Jinja returns the whole block at a time, and that's a `{% raw %} ... {% endraw %}` section.